packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/functions/SuspendFunctionComponent.kt

```diff
@@ -0,0 +1,56 @@
+package expo.modules.kotlin.functions
+
+import com.facebook.react.bridge.ReadableArray
+import expo.modules.kotlin.ModuleHolder
+import expo.modules.kotlin.Promise
+import expo.modules.kotlin.exception.CodedException
+import expo.modules.kotlin.exception.FunctionCallException
+import expo.modules.kotlin.exception.UnexpectedException
+import expo.modules.kotlin.exception.exceptionDecorator
+import expo.modules.kotlin.types.AnyType
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.isActive
+import kotlinx.coroutines.launch
+import java.lang.ref.WeakReference
+import kotlin.jvm.Throws
+
+/**
+ * We can't construct a `SuspendFunctionComponent` in the build phase, cause it has to have access to module coroutine scope.
+ * So we create another builder to store need information and build it later - during the holder initialization.
```

Comment:
```suggestion
 * We can't construct a `SuspendFunctionComponent` in the build phase, because it has to have access to module coroutine scope.
 * So we create another builder to store needed information and build it later - during the holder initialization.
```